src/android/CleverTapPlugin.java

```diff
@@ -229,18 +222,14 @@ public void run() {
 
         // not required for Android here but handle as its in the JS interface
         else if (action.equals("registerPush")) {
-            result = new PluginResult(PluginResult.Status.NO_RESULT);
-            result.setKeepCallback(true);
-            callbackContext.sendPluginResult(result);
+            sendPluginResult(callbackContext, PluginResult.Status.NO_RESULT);
             return true;
         } else if (action.equals("setPushTokenAsString")) {
             final String token = args.getString(0);
             cordova.getThreadPool().execute(new Runnable() {
                 public void run() {
                     cleverTap.pushFcmRegistrationId(token, true);
-                    PluginResult _result = new PluginResult(PluginResult.Status.NO_RESULT);
-                    _result.setKeepCallback(true);
-                    callbackContext.sendPluginResult(_result);
+                    sendPluginResult(callbackContext, PluginResult.Status.NO_RESULT);
```

Comment:
Can we have the whole `cordova.getThreadPool().execute(new Runnable() {` within `sendPluginResult` ? It seems to be used almost everywhere. Check why sometimes it is not used and decide on whether to have it a as param to the `sendPluginResult` `runInBackground`

Reply:
`cordova.getThreadPool().execute(new Runnable() { `

is also used to make a call to native APIs, hence it cannot be directly extracted out into the sendPluginResult?